File: 프로그래머스/더맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> queue = new PriorityQueue<Integer>();
        
        for(Integer i: scoville){
            queue.add(i);
        }
        
        while (queue.peek() < K){
            if (queue.size() == 1){
                return -1;
            }
            
            int x = queue.poll();
            
            if(x < K){
                int y = queue.poll();
                queue.add(x + 2 * y);
                answer++;
            }
            
        }
        return answer;
    }
}

File: 프로그래머스/더맵게.py
import heapq

def solution(scoville, K):
    heap = []
    
    for scov in scoville:
        heapq.heappush(heap, scov)
    answer = 0
    
    
    while heap[0] < K:
        
        if len(heap) == 1:
            return -1
        
        x = heapq.heappop(heap)
        
        if x < K:
            y = heapq.heappop(heap)
            
            heapq.heappush(heap, x + y * 2)
            answer += 1
            
            
            
    return answer